test/ts_test/src/service.ts

```diff
@@ -21,4 +23,11 @@ describe("ts service", () => {
     assert.strictEqual(simple_service_pb_service.SimpleService.DoStream.requestType, simple_service_pb.StreamRequest);
     assert.strictEqual(simple_service_pb_service.SimpleService.DoStream.responseType, external_child_message_pb.ExternalChildMessage);
   });
+  it("should not output imports for namespaces that are not used in the service definition", () => {
```

Comment:
A better solution is to pull https://github.com/improbable-eng/ts-protoc-gen/blob/0135ec77326fb35535af0a7751d48c83eecea99e/src/ts/fileDescriptorTSServices.ts#L26-L34
into a separate method which accepts the `fileDescriptor`. We can then mock the `fileDescriptor` and assert the output string doesn't contain the import. 

I realize most of that testing style isn't in place so your solution will do for now and we can address it later.

Reply:
Agree this is a far better solution and happy to make a subsequent PR to try to improve the tests.